### PR TITLE
CTCP-4632: Additional reference number page

### PR DESCRIPTION
**Pull request by justega247**

### CODE REVIEW

File: app/pages/houseConsignment/index/items/additionalReference/AdditionalReferenceNumberPage.scala [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import pages.QuestionPage
+import pages.sections.houseConsignment.index.items.additionalReference.AdditionalReferenceSection
+import play.api.libs.json.JsPath
+
+case class AdditionalReferenceNumberPage(houseConsignmentIndex: Index, itemIndex: Index, additionalReferenceIndex: Index) extends QuestionPage[String] {
+
+  override def path: JsPath =
+    AdditionalReferenceSection(houseConsignmentIndex, itemIndex, additionalReferenceIndex).path \ toString
+
+  override def toString: String = "number"
```
Comment (joecefnhedog) [inline app/pages/houseConsignment/index/items/additionalReference/AdditionalReferenceNumberPage.scala:29]: can this be referenceNumber to match the 43

**Review by joecefnhedog — APPROVED** [review]
LGTM just a minor comment

File: app/controllers/houseConsignment/index/items/additionalReference/AdditionalReferenceNumberController.scala [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    additionalReferenceIndex: Index,
+    mode: Mode
+  )(implicit request: MandatoryDataRequest[_]): Future[Result] =
+    for {
+      updatedAnswers <- Future.fromTry(
+        request.userAnswers.set(AdditionalReferenceNumberPage(houseConsignmentIndex, itemIndex, additionalReferenceIndex), value)
+      )
+      _ <- sessionRepository.set(updatedAnswers)
+    } yield Redirect(navigator.nextPage(AdditionalReferenceNumberPage(houseConsignmentIndex, itemIndex, additionalReferenceIndex), mode, request.userAnswers))
+}
```
Comment (prashanthmamidi) [inline app/controllers/houseConsignment/index/items/additionalReference/AdditionalReferenceNumberController.scala:116]: In the Redirect(), whether we need to use `request.userAnswers` or `updatedAnswers`

Comment (justega247) [reply app/controllers/houseConsignment/index/items/additionalReference/AdditionalReferenceNumberController.scala:116]: It's the request.userAnswers we use.

**Review by joecefnhedog — APPROVED** [review]

**Pull request merged**